js/search/search.js

```diff
@@ -519,160 +518,61 @@ class Search {
         return out;
     }
 
-    /**
-     * Validate the index folder exist or not
-     * @param {String} file
-     * @returns {*}
-     */
-    static indexValidator(file) {
-        let data;
-        let result = childProcess.execFileSync(INDEX_VALIDATOR, [file]).toString();
-        try {
-            data = JSON.parse(result);
-            if (data.status === 'OK') {
-                return data;
-            }
-            log.send(logLevels.ERROR, 'Unable validate index folder');
-            return new Error('Unable validate index folder')
-        } catch (err) {
-            throw new Error(err);
-        }
-    }
+}
 
-    /**
-     * Removing all the folders and files inside the data folder
-     * @param location
-     */
-    static deleteIndexFolders(location) {
-        if (fs.existsSync(location)) {
-            fs.readdirSync(location).forEach((file) => {
-                let curPath = location + "/" + file;
+function clearSearchData() {
+    function removeFiles(filePath) {
+        if (fs.existsSync(filePath)) {
+            fs.readdirSync(filePath).forEach((file) => {
+                let curPath = filePath + "/" + file;
                 if (fs.lstatSync(curPath).isDirectory()) {
-                    Search.deleteIndexFolders(curPath);
+                    removeFiles(curPath);
                 } else {
                     fs.unlinkSync(curPath);
                 }
             });
-            fs.rmdirSync(location);
+            fs.rmdirSync(filePath);
```

Comment:
won't work if there are files inside the dir

Reply:
![screen shot 2018-06-08 at 8 18 54 pm](https://user-images.githubusercontent.com/596478/41164603-349127be-6b59-11e8-9d7c-a89731b569ee.png)

All the files inside the folder will be deleted before deleting the folder.